packages/block-library/src/template-part/editor.scss

```diff
@@ -77,6 +80,33 @@
 	}
 }
 
+.wp-block-template-part__label-container {
+	position: absolute;
+	bottom: 100%;
+	width: 100%;
+
+	.wp-block-template-part__label-layout {
+		display: flex;
+		justify-content: flex-start;
+	}
+
+	.wp-block-template-part__label-content {
+		display: flex;
+		align-items: center;
+		border: $border-width solid $gray-600;
+		border-bottom: none;
+		border-top-right-radius: $radius-block-ui;
+		border-top-left-radius: $radius-block-ui;
+		padding: 0 6px;
+	}
+
+	.wp-block-template-part__label-text {
+		font-size: $default-font-size;
+		text-transform: uppercase;
```

Comment:
I don't think we need to make this all uppercase; We don't do that elsewhere (do we?), and uppercase words like this can be more difficult to read (you _lose_ the letter shapes).

Reply:
> uppercase words like this can be more difficult to read (you lose the letter shapes)

Looking at your examples with Twenty Nineteen, `UNTITLED TEMPLATE PART` _does_ seem difficult to read.

> We don't do that elsewhere (do we?)

This is a very good point. 👍

Reply:
Pushed changes to switch the label to title case